src/StatsTools/convenience.jl

```diff
@@ -130,3 +86,150 @@ function to_measurement(p::MonteCarloMeasurements.Particles)
+end
+function _errs(::Nothing, m::Measurement, n, name)
+    σ = uncertainty(m)
+    return (; Symbol(name) => value(m), Symbol(name, :_l) => n*σ, Symbol(name, :_u) => n*σ)
+end
+function _errs(p, m::Measurement, _, name)
+    d = Normal(value(m), uncertainty(m))
+    cp1 = (1-p)/2
+    cp2 = 1-cp1
+    q1, q2 = quantile(d, cp1), quantile(d, cp2)
+    return (;
+        Symbol(name) => value(m),
+        Symbol(name, :_l) => value(m) - q1,
+        Symbol(name, :_u) => q2 - value(m),
+    )
+end
+function _errs(::Nothing, m::AbstractParticles, n, name)
+    p = erf(n/√2)
+    return _errs(p, m, n, name)
+end
+function _errs(p, m::AbstractParticles, _, name)
+    cp1 = (1-p)/2
+    cp2 = 1-cp1
+    q1, q2, q3 = pquantile(m, (cp1, 0.5, cp2))
+    return (;Symbol(name) => q2, Symbol(name, :_l) => q2 - q1, Symbol(name, :_u) => q3 - q2)
+end
+function val_and_errs(r::BlockingResult; kwargs...)
+    return val_and_errs(measurement(r.mean, r.err); kwargs...)
+end
+val_and_errs(r::RatioBlockingResult; kwargs...) = val_and_errs(r.ratio; kwargs...)
+
+"""
+    errs(x; n=1, p=nothing, name=:err) -> (; err_l, err_u)
+Return the lower and upper error bar for the uncertain value `x`.
+
+See [`val_and_errs`](@ref).
+"""
+function errs(args...; name=:err, kwargs...)
+    _, err_l, err_u = val_and_errs(args...; kwargs...)
+    return (; Symbol(name, :_l) => err_l, Symbol(name, :_u) => err_u)
+end
+
+"""
+    to_nt(x; n=1, p=nothing, name=:val) -> nt::NamedTuple
+Return a named tuple with value and error bars (see [`val_and_errs`](@ref)) as well
+as additional numerical fields relevant for `x`.
+
+**Example:**
+```jldoctest
+julia> results = [blocking_analysis(i:0.1:2i+20) for i in 1:3]; # mock results
+
+julia> df = to_nt.(results, name=:res)|>DataFrame
+3×7 DataFrame
+ Row │ res      res_l    res_u    res_err_err  res_p_cov  res_k  res_blocks
+     │ Float64  Float64  Float64  Float64      Float64    Int64  Int64
+─────┼──────────────────────────────────────────────────────────────────────
+   1 │    11.5  1.7282   1.7282      0.352767    2.98667      5          13
+   2 │    13.0  1.7282   1.7282      0.352767    2.98667      5          13
+   3 │    14.5  1.78885  1.78885     0.350823    3.2          5          14
+```
+```julia-repl
+julia> rbs = ratio_of_means(1 .+sin.(1:0.1:11),2 .+sin.(2:0.1:12)); # more mock results
+
+julia> [to_nt(rbs),]|>DataFrame
+1×9 DataFrame
+ Row │ val       val_l      val_u      val_f     val_σ_f    val_δ_y    val_k  val_blocks  val_success
+     │ Float64   Float64    Float64    Float64   Float64    Float64    Int64  Int64       Bool
+─────┼────────────────────────────────────────────────────────────────────────────────────────────────
+   1 │ 0.581549  0.0925669  0.0812292  0.560532  0.0875548  0.0875548      4          12         true
+
+```
+
+See [`val_and_errs`](@ref), [`val`](@ref), [`errs`](@ref), [`BlockingResult`](@ref),
+[`RatioBlockingResult`](@ref).
+"""
+to_nt(args...; kwargs...) = val_and_errs(args...; kwargs...)
```

Comment:
I'm not sure I like the name `to_nt`. Maybe it could be turned around to be a method of `Base.NamedTuple`, but that would prevent us from having the first method.

Reply:
I agree, that would be cleaner.

Maybe even a method for 
```
DataFrame(v::Vector{BlockingResult})
```
?

Reply:
Good idea. An even fancier method would be to make `Vector{BlockingResult}` Tables.jl compliant. That way, `DataFrame(...)` will work and it will also work with other things in the data ecosystem.

Reply:
Sounds good, but I guess we won't be able to pass through kwargs for a name prefix, or the level of the confidence interval. Maybe this information could be added to the `BlockingResult` and `RatioBlockingResult` structs, though - maybe using `Ref`s? What do you think?

Reply:
I see, I missed the fact you need `kwargs...`. I'm not sure what the best solution is. Having the name and intervals in the `RatioBolckingResult` is a bit annoying if you want to change it later. Maybe something like `to_nt` is a good option. A vector of named tuples is also a Tables.jl table.

If we decide to go with keeping the data in the `struct`, I would prefer having some kind of setter functions that return a new instance with a different name/intervals (rather than a `Ref`).

Reply:
I added methods (with keyword arguments) for `NamedTuple` for `BlockingResult` and `RatioBlockingResult`. For `Particles` and `Measurement` you can use `val_and_errs` instead. Do you think that there is an issue with adding these weird keyword arguments to a `Base` function?

Funnily, `DataFrame(Vector{BlockingResult})` already works out of the box:
```
julia> r = blocking_analysis(smoothen(randn(2^10), 2^5));

julia> DataFrame([r,])
1×6 DataFrame
 Row │ mean        err        err_err     p_cov       k      blocks 
     │ Float64     Float64    Float64     Float64     Int64  Int64  
─────┼──────────────────────────────────────────────────────────────
   1 │ -0.0155974  0.0341049  0.00433133  0.00116314      6      32
```
I'm surprised and hadn't expected that. You cannot change names or intervals this way, though.

We could try to make it work this way for `RatioBlockingResult` as well with default name and confidence interval. However, one could always go through `NamedTuple` (or through `val_and_errs` if you are only interested in the error bars), which is not too bad I would think.

Reply:
There's no problem with adding weird arguments as long as it dispatches on a type you own (which this does). It's interesting that it works with Tables like that. I think it uses `propertynames` of the types as columns.